src/lib/common/tag.h

```diff
@@ -30,6 +30,22 @@
 #include "common/Format.h"
 
 
+// Macros to be used for JSON rendering
+
+#define JSON_PROP(name) std::string("\"" + std::string(name) + "\":")
+
+#define JSON_STR_LAST(value) std::string("\"" + std::string(value) + "\"")
+#define JSON_STR(value) std::string(JSON_STR_LAST(value) + ",")
+
+#define JSON_NUMBER_LAST(value) std::string(value)
+#define JSON_NUMBER(value) std::string(JSON_NUMBER_LAST(value) + ",")
+
+#define JSON_VALUE_LAST(name,value) std::string(JSON_PROP(name) + JSON_STR_LAST(value))
+#define JSON_VALUE(name,value) std::string(JSON_VALUE_LAST(name,value) + ",")
+
+#define JSON_VALUE_NUMBER_LAST(name,value) std::string(JSON_PROP(name) + JSON_NUMBER_LAST(value))
+#define JSON_VALUE_NUMBER(name,value) std::string(JSON_VALUE_NUMBER_LAST(name,value) + ",")
+
 
```

Comment:
I would not encourage to use the macros with the comma incorporated.
It only adds complexity ... (IMHO)